crates/walrus-sui/src/client/retry_client.rs

```diff
@@ -1202,6 +1202,19 @@ impl From<tonic::Status> for RetriableClientError {
     }
 }
 
+impl RetriableClientError {
+    fn is_eligible_for_fallback(&self, next_checkpoint: u64) -> bool {
+        match self {
+            Self::RpcError(rpc_error) if rpc_error.status.code() == tonic::Code::NotFound => {
+                rpc_error
+                    .checkpoint_seq_num
```

Comment:
question: checkpoint_seq_num is the same as the next_checkpoint, is it? Should we use `rpc_error.status.checkpoint_height()` here?

Reply:
`checkpoint_height()` is a function on status? I am not sure, my rust analyzer is so slow - I can't even tell :(. In any case, i think the end result is the same - so will keep it

Reply:
My main question is, `rpc_error. checkpoint_seq_num` is actually the checkpoint height returned from the fullnode?

It seems that [the rpc_error.checkpoint_seq_num is always set to be the checkpoint number that this request tries to fetch](https://github.com/MystenLabs/walrus/blob/f3b8641456b0a56e644fa82c90bc42f6c46d8330/crates/walrus-sui/src/client/retry_client.rs#L1352), which makes this check here always a `false`. Am I reading the code correctly?